test/govuk_component/button_test.rb

```diff
@@ -55,4 +55,13 @@ def component_name
     render_component(text: "Start now", rel: "nofollow preload")
     assert_select ".pub-c-button[rel='nofollow preload']", text: "Start now"
   end
+
+  test "renders margin bottom class correctly" do
+    render_component(text: "Submit")
+    assert_select ".pub-c-button", text: "Submit"
```

Comment:
I'm not sure this line is necessary, if the test fails at this point it doesn't prove anything to do with the margin class but a different fail in the component, I'd have thought the selector test on its own was enough.

Reply:
This ensures the component renders as expected, with the next assertion making sure that it is then applied the right class.

Without this assertion you could in theory only render the modifier class, or this option could break how the text renders etc.

Reply:
I'd argue that your previous tests test that the component renders ok and it should fail earlier against relevant cases, but I'm not going to argue that hard :) Just personal preference, really.

Reply:
If I were to introduce behaviour that removed the text output *only* when passing the 'margin_bottom' option, this test would fail.

If I didn't test this, it would not fail.

Ideally we'd be doing a form of [snapshot testing](https://facebook.github.io/jest/docs/en/snapshot-testing.html)